src/StackExchange.Redis/ResultProcessor.cs

```diff
@@ -1083,6 +1087,24 @@ protected override bool SetResultCore(PhysicalConnection connection, Message mes
             }
         }
 
+        private sealed class NullableDoubleArrayProcessor : ResultProcessor<double?[]>
+        {
+            protected override bool SetResultCore(PhysicalConnection connection, Message message, in RawResult result)
+            {
+                switch (result.Type)
+                {
+                    case ResultType.Integer:
+                    case ResultType.SimpleString:
+                    case ResultType.BulkString:
+                    case ResultType.MultiBulk:
```

Comment:
I believe only `MultiBulk` is allowed here (per https://redis.io/commands/zmscore/), which means we can simplify to `if (result.Type == ResultType.MultiBulk && !result.IsNull)` (though I don't _think_ the `IsNull` case can happen, let's be safe). If we're asserting the result of `GetItemsAsDoubles` is not null, we should ensure that either way (with the check).

Reply:
Updated with the simplified check.